### DISCUSSION TITLE
Doubts about how to use the library.

### DISCUSSION THREAD

**Discussion by cirobruno** [khoih-prog/ESPAsync_WiFiManager#67] [Q&A]
Dear Developers

Congratulations for the development of your library, as well as its documentation. It's really a high grade professional resource you've kindly offered for free for the whole community. Thank you.

I'd like to ask some questions about its usage. Maybe I'll have to ask them during my learning process.

I've just included the version [https://github.com/khoih-prog/ESPAsync_WiFiManager/blob/master/examples/Async_AutoConnectWithFeedback/Async_AutoConnectWithFeedback.ino](url) in a code I've been writing.

1) The code originally contained credentials for logging in my office's network already. Then, after replacing the original library by ESPAsync_WiFiManager code, the device has automatically logged in the same network. Has the ESPAsync_WiFiManager library got this credentials from some piece of code for doing so?

2) Since the code logs in automatically, how can I force it to search for another network to log in? The documentations mentions a button to be pressed. Where is this button declared in the code? Which function is it supposed to call?

3) Does the library contain support for FTP?

3) Does the library contain support for NTP?

Thank you.
Regards,
Ciro Bruno.

**Comment by cirobruno** [comment]
Making some progress...

I found this among the instructions in setup() block:

//reset settings - for testing
// ESPAsync_wifiManager.resetSettings();

Then I’ve uncommented it and flashed it again.

Now, during startup it reports via Serial:
<code>
Starting Async_AutoConnectWithFeedBack using SPIFFS on TTGO LoRa32-OLED v2.1.6

ESPAsync_WiFiManager v1.9.3

[WM] RFC925 Hostname = AutoConnectWithFeedBack

[WM] setSTAStaticIPConfig

[WM] Set CORS Header to : Your Access-Control-Allow-Origin

ESP Self-Stored: SSID = 0, Pass = 0

[WM] * Add SSID = 0 , PW = 0

Got ESP Self-Stored Credentials. Timeout 120s for Config Portal

[WM] LoadWiFiCfgFile

[WM] OK

[WM] stationIP = 0.0.0.0 , gatewayIP = 0.0.0.0

[WM] netMask = 0.0.0.0

[WM] dns1IP = 0.0.0.0 , dns2IP = 0.0.0.0

Got stored Credentials. Timeout 120s for Config Portal

Current Timezone is not set. Enter Config Portal to set.

ConnectMultiWiFi in setup

[WM] ConnectMultiWiFi with :

[WM] * Flash-stored Router_SSID = 0 , Router_Pass = 0

[WM] * Add SSID = 0 , PW = 0

[WM] Connecting MultiWifi…

[E][WiFiMulti.cpp:191] run(): [WIFI] no matching wifi found!

[WM] WiFi not connected

ets Jun 8 2016 00:22:57

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
…
</code>

Now it keeps reseting time after time.
Shouldn’t the Config Portal be opened and ask for network credentials?

What should I do now?
Thank you.
Regards, Ciro.

**Comment by khoih-prog** [comment]
Please read carefully the section [**So, how it works?**](https://github.com/khoih-prog/ESPAsync_WiFiManager#so-how-it-works).

You have to enter the **Config Portal** in order to input the correct customs Credentials, according to your local network, so that the system can work. Otherwise, the ESP32 just tries to use the stored (and **wrong Credentials: SSID/Password from previous network**) and can't connect => **reset repetitively**

![image](https://user-images.githubusercontent.com/57012152/136484560-9bef00a5-d530-44d8-92fd-00821f9cf39a.png)


It's better to use more complex examples, such as [**Async_ConfigOnDoubleReset**](https://github.com/khoih-prog/ESPAsync_WiFiManager/tree/master/examples/Async_ConfigOnDoubleReset), then use Double Reset to enter Config Portal.

Without entering Config Portal, you can't connect to any new network.

**Reply by cirobruno** [reply]
Thank you for answering.

The device's AP wasn't available until I forced it. I had to make it ignore "initialConfig" value and replace it by "true".

Please take a look at my message bellow.

(It looks like we've been writing at the same time)

**Comment by cirobruno** [comment]
It looks like there's a trouble with some condition.

In the part of code that checks `initialConfig`:
<code>
  if (initialConfig)

  {

    Serial.println(F("We haven't got any access point credentials, so get them now"));

    Serial.print(F("Starting configuration portal @ "));

...
</code>

I've forced the condition by:
<code>
 // if (initialConfig)

if (true)

  {

    Serial.println(F("We haven't got any access point credentials, so get them now"));

    Serial.print(F("Starting configuration portal @ "));

...
</code>

Doing this I could be able to find "ESP_xxxxxxxx_AutoConnectAP" network, to open Config Portal and to input my office's network credentials.

Then, I've replaced the original "if" and flashed again.

After that, when it starts, it doesn't enter AP mode anymore, asks for connection in my office's network. It looks like my router gives it a IP address (picture bellow): 192.168.4.106 (picture bellow).
![image](https://user-images.githubusercontent.com/11180616/136484284-def81b15-90d6-48a5-9c3a-39a18ce69aea.png)

Despite this, the device tries to use 192.168.2.232

<code>
Starting Async_AutoConnectWithFeedBack using SPIFFS on TTGO LoRa32-OLED v2.1.6

ESPAsync_WiFiManager v1.9.3

[WM] RFC925 Hostname = AutoConnectWithFeedBack

[WM] setSTAStaticIPConfig

[WM] Set CORS Header to :  Your Access-Control-Allow-Origin

ESP Self-Stored: SSID = WiFi-Home 4, Pass = gracamello

[WM] * Add SSID =  WiFi-Home 4 , PW =  gracamello

Got ESP Self-Stored Credentials. Timeout 120s for Config Portal

[WM] LoadWiFiCfgFile 

[WM] OK

[WM] stationIP = 192.168.2.232 , gatewayIP = 192.168.2.1

[WM] netMask = 255.255.255.0

[WM] dns1IP = 192.168.2.1 , dns2IP = 8.8.8.8

Got stored Credentials. Timeout 120s for Config Portal

[WM] Current TZ_Name = America/Argentina/Buenos_Aires , TZ =  <-03>3

[WM] * Add SSID =  WiFi-Home 4 , PW =  gracamello

ConnectMultiWiFi in setup

[WM] ConnectMultiWiFi with :

[WM] * Flash-stored Router_SSID =  WiFi-Home 4 , Router_Pass =  gracamello

[WM] * Add SSID =  WiFi-Home 4 , PW =  gracamello

[WM] * Additional SSID =  WiFi-Home 4 , PW =  gracamello

[WM] Connecting MultiWifi...

[WM] WiFi connected after time:  1

[WM] SSID: WiFi-Home 4 ,RSSI= -52

[WM] Channel: 11 ,IP address: 192.168.2.232

After waiting 9.90 secs more in setup(), connection result is connected. Local IP: 192.168.2.232

[WM] freeing allocated params!
</code>

What to do now?

**Comment by khoih-prog** [comment]
You have to read more carefully the instructions and understand the operation before moving forward.

Don't waste your time going blindly, using trials-and-errors methods.

Sorry I don't have time now to lead you step-by-step. If necessary, ask in some forums

**Reply by cirobruno** [reply]
I've just copied and pasted the example code into mine.

In my opinion, the hardware button would be very useful to fix some network credentials issues.
